File: src/#1431/solution.go
package solution

func kidsWithCandies(candies []int, extraCandies int) []bool {
	maxNum := 0
	for _, v := range candies {
		if v > maxNum {
			maxNum = v
		}
	}
	res := make([]bool, len(candies))
	for i, v := range candies {
		if v+extraCandies >= maxNum {
			res[i] = true
		} else {
			res[i] = false
		}
	}
	return res
}

File: src/#1431/solution.js
var kidsWithCandies = function (candies, extraCandies) {
  let maxNum = Math.max(...candies);
  let res = new Array(candies.length);

  candies.forEach((v, i) => {
    res[i] = v + extraCandies >= maxNum;
  });

  return res;
};
